src/Albany_Application.cpp

```diff
@@ -613,7 +584,8 @@ Application::finalSetUp(
       initial_guess,
       paramLib,
       disc,
-      comm));
+      comm,
+      num_params_));
```

Comment:
Wouldn't this deduction duplicate the code in calcTangentDerivDimension in Albany::Application?  I would get the # params from the PL but it seems w/ all the different parameter list options we support (old, new, etc.) it would be on the order of 15 lines of code, that would get duplicated in various places.  You know better than me the different ways to specify the params.  Do you have a suggestion how to succinctly infer them from the PL other than this: https://github.com/SNLComputation/Albany/blob/observer-trans-sens/src/Albany_Application.cpp#L2661-L2680 ?  If so, I'm happy to switch to that.  I thought passing the parameter was the simplest way around repeating all that code to parse the PL but maybe I'm overlooking something.

Reply:
No, I agree you would have to use that code (which would be simpler once we get rid of the old syntax). It's really to avoid passing num_params in all those places. Anyway, I'm fine with the current implementation. It is probably not worth the effort to remove that parameter.

Reply:
I agree with you 100% that it would be cleaner to pass the num_params around I just couldn't think of a clean way to avoid it.  I think after I discovered that there are so many ways the parameters can be specified, I was even more hesitant to try to infer things from the PL having to do with them...  Anyway, if you're OK with things as it I think I'll keep it, but I'll give it a day or two to see if anyone else, e.g. Glen (another reviewer), has better suggestions.  How about that as a compromise?

Reply:
Sounds good!